### PR TITLE
Login-UI with reset component

### PR DESCRIPTION
**Pull request by skanthed**
I have closed the old PR. This is the updated PR with comments and postman Integration in the login component.

### CODE REVIEW

File: src/app/LoginPage/LoginPage.tsx [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
 } from '@patternfly/react-core';
+import { LoginForm } from './LoginPageForm';
 import { ExclamationCircleIcon } from '@patternfly/react-icons';
 import * as OAuth from 'oauth2-client-js';
 
 import { Role } from '..';
 import mocLogo from './moc_logo.png';
+import { ResetPasswordForm } from './ResetPasswordForm';
+import axios from 'axios';
+import { debug } from 'console';
```
Comment (gagansk) [inline src/app/LoginPage/LoginPage.tsx:15]: Remove this

File: src/app/LoginPage/LoginPage.tsx [hunk 2]
```diff
@@ -35,21 +38,47 @@ type LoginState = {
...
+      api: 'https://cce4c6f9-ed0c-4bd9-94c7-6f174f7ae024.mock.pstmn.io/login',
+      isValidUser: true,
+      isValidOldPassword: undefined,
+      isResetPasswordProcessed: undefined,
+      oldPassword: "",
+      newPassword: "",
+      confirmPassword: "",
     };
+
+
```
Comment (gagansk) [inline src/app/LoginPage/LoginPage.tsx]: Remove additional new line

File: src/app/LoginPage/LoginPage.tsx [hunk 3]
```diff
@@ -89,22 +128,94 @@ class LoginPage extends React.Component<LoginProps, LoginState> {
     this.setState({ password: passwordInput });
   }
 
+  validateOldPassword = () => {
+    let apiUrl = "https://cce4c6f9-ed0c-4bd9-94c7-6f174f7ae024.mock.pstmn.io/validate_old_password?old_password="
+    axios
+      .get(apiUrl + this.state.oldPassword)
+      .then(res => {
+        debugger
```
Comment (gagansk) [inline src/app/LoginPage/LoginPage.tsx]: Take out debugger

File: src/app/LoginPage/LoginPage.tsx [hunk 4]
```diff
@@ -89,22 +128,94 @@ class LoginPage extends React.Component<LoginProps, LoginState> {
...
+    let apiUrl = "https://cce4c6f9-ed0c-4bd9-94c7-6f174f7ae024.mock.pstmn.io/reset"
+    const credential = {
+      oldPassword: this.state.oldPassword,
+      newPassword: this.state.newPassword
+    };
+    this.setState({ isValidOldPassword: undefined })
+    axios
+      .post(apiUrl, { credential })
+      .then(res => {
+        debugger
```
Comment (gagansk) [inline src/app/LoginPage/LoginPage.tsx]: Same. take out the debugger.

File: src/app/LoginPage/LoginPage.tsx [hunk 5]
```diff
@@ -158,13 +299,16 @@ class LoginPage extends React.Component<LoginProps, LoginState> {
...
+        {/* {this.state.isLoginPage ? <div style={{ color: "red" }} > {this.state.isValidUser == false ? "Invalid Credentials" : null}</div> : null}*/}
+        {(this.state.isLoginPage == false && this.state.isValidOldPassword != undefined) ? <div style={{ color: this.state.isValidOldPassword ? "green" : "red" }} > {this.state.isValidOldPassword == true ? "Verified Old Password" : "Invalid Old Password"}</div> : null}
+        {(this.state.isLoginPage == false && this.state.isResetPasswordProcessed != undefined) ? <div style={{ color: this.state.isResetPasswordProcessed ? "green" : "red" }} > {this.state.isResetPasswordProcessed ? "Updated your password successfully" : "Error is occured. Please contact admin!"}</div> : null}
+      </PatternflyLoginPage >
     );
   }
 }
 
-export { LoginPage };
+export { LoginPage };
```
Comment (gagansk) [inline src/app/LoginPage/LoginPage.tsx]: Add a new line

File: src/app/LoginPage/LoginPageForm.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                </FormGroup>
+            )}
+            <ActionGroup>
+                <Button variant="primary" type="submit" onClick={onLoginButtonClick} isBlock isDisabled={isLoginButtonDisabled}>
+                    {loginButtonLabel}
+                </Button>
+                <a href='javascript:void(0)' onClick={onResetButtonClick} style={{ paddingLeft: "35%" }}>Reset Password</a>                
+            </ActionGroup>
+        </Form>
+    );
```
Comment (gagansk) [inline src/app/LoginPage/LoginPageForm.tsx]: Add a new line

File: src/app/LoginPage/ResetPasswordForm.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+                />
+            </FormGroup>
+            <ActionGroup>
+                <Button variant="primary" type="submit" onClick={onLoginButtonClick} isBlock isDisabled={isResetButtonDisabled}>
+                    {resetButtonLabel}
+                </Button>
+                <a href='javascript:void(0)' onClick={onResetButtonClick} style={{ paddingLeft: "40%" }}>Login</a>
+            </ActionGroup>
+        </Form>
+    );
```
Comment (gagansk) [inline src/app/LoginPage/ResetPasswordForm.tsx]: add a new line

File: src/app/LoginPage/LoginPage.tsx [hunk 8]
```diff
@@ -89,22 +128,94 @@ class LoginPage extends React.Component<LoginProps, LoginState> {
...
+    //   role = Role.ADMIN;
+    //   localStorage.setItem('login', 'ADMIN');
+    // } else if ((username === 'developer1' && password === 'developer1pass') || (username === 'developer2' && password === 'developer2pass')) {
+    //   role = Role.DEVELOPER;
+    //   localStorage.setItem('login', 'DEVELOPER');
+    // } else {
+    //   role = Role.NONE;
+    //   this.setState({ ...this.state, error: "Invalid Username/Password" })
+    // }
+    // this.props.setRole(role);
```
Comment (gagansk) [inline src/app/LoginPage/LoginPage.tsx:213]: You aren't checking for the password match and hence, the application is taking every input as a valid password. Please review this.

**Review by gagansk — CHANGES_REQUESTED** [review]
Password validation is broken. Please fix this.

File: src/app/LoginPage/LoginPage.tsx (hunk 2 above)
Comment (skanthed) [reply src/app/LoginPage/LoginPage.tsx]: Done

File: src/app/LoginPage/LoginPage.tsx (hunk 3 above)
Comment (skanthed) [reply src/app/LoginPage/LoginPage.tsx]: Yes. I removed it from everywhere.

File: src/app/LoginPage/LoginPage.tsx (hunk 8 above)
Comment (skanthed) [reply src/app/LoginPage/LoginPage.tsx:213]: We will check password after retrieving values from the database.

**Review by gagansk — APPROVED** [review]
LGTM

**Pull request merged**